gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnAutoScalingManager.java

```diff
@@ -219,13 +226,25 @@ private String getInuseParticipantForHelixPartition(JobContext jobContext, int p
       return null;
     }
 
+
+    private String getParticipantInInitStateForHelixPartition(JobContext jobContext, int partition) {
+      if (jobContext.getPartitionState(partition).equals(TaskPartitionState.INIT)) {
+        log.info("Helix task {} is in {} state",
+            jobContext.getTaskIdForPartition(partition), jobContext.getPartitionState(partition));
+        return jobContext.getAssignedParticipant(partition);
+      }
+
+      return null;
+    }
+
     /**
      * Iterate through the workflows configured in Helix to figure out the number of required partitions
      * and request the {@link YarnService} to scale to the desired number of containers.
      */
     @VisibleForTesting
     void runInternal() {
       Set<String> inUseInstances = new HashSet<>();
+      Set<String> instancesInInitState = new HashSet<>();
```

Comment:
Let's add a comment, that instancesInInitState contains only those containers where task is assigned, not all the helix containers in init state. We can change the variable name to signify that.